src/es/templescanesp/src/eu/kanade/tachiyomi/extension/es/templescanesp/TempleScanEsp.kt

```diff
@@ -28,25 +31,68 @@ class TempleScanEsp :
     ConfigurableSource {
 
     override val baseUrl get() = preferences.prefBaseUrl
+    private val preferences = getPreferences {
+        val domain = this.getString(DEFAULT_BASE_URL_PREF, null)
+        if (domain != super.baseUrl) {
+            this.edit()
+                .putString(BASE_URL_PREF, super.baseUrl)
+                .putString(DEFAULT_BASE_URL_PREF, super.baseUrl)
+                .apply()
+        }
+        Unit
+    }
 
     private val fetchedDomainUrl: String by lazy {
         if (!preferences.fetchDomainPref()) return@lazy preferences.prefBaseUrl
+
         try {
             val initClient = network.cloudflareClient
-            val headers = super.headersBuilder().build()
-            val document = initClient.newCall(GET("https://templescanesp.net", headers)).execute().asJsoup()
+            val headers = super.headersBuilder()
+                .add("apikey", SUPABASE_API_KEY)
+                .add("Accept", "application/json")
+                .build()
+
+            initClient.newCall(GET(SUPABASE_URL, headers)).execute().use { resp ->
+                if (resp.isSuccessful) {
+                    val body = resp.body?.string().orEmpty()
+                    val value = try {
+                        json.parseToJsonElement(body).jsonArray.first().jsonObject["value"]!!.jsonPrimitive.content
+                    } catch (_: Exception) {
+                        null
+                    }
+
+                    if (!value.isNullOrBlank()) {
+                        val detected = value.trimEnd('/')
+                        val newDomain = if (detected.startsWith("http")) detected else "https://$detected"
+
+                        if (shouldUpdatePref()) {
+                            preferences.prefBaseUrl = newDomain
+                        }
+
+                        return@use newDomain
+                    }
+                }
+                Unit
+            }
+
+            val htmlHeaders = super.headersBuilder().build()
+            val document = initClient.newCall(GET("https://templescanesp.net", htmlHeaders)).execute().asJsoup()
```

Comment:
Just remove the html fallback, if they break it we can just remove the domain updater